File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>prefix;
        int sum=0,count=0;
        prefix[0]=1;
        for(auto &num:nums){
            sum+=num;
            count+=prefix[sum-k];
            prefix[sum]++;
        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefix=defaultdict(int)
        count=0
        sum=0
        prefix[0]=1

        for num in nums:
            sum+=num
            count+=prefix[sum-k]
            prefix[sum]+=1
        
        return count